core-bundle/src/Filesystem/Dbafs/Dbafs.php

```diff
@@ -197,6 +197,9 @@ public function setExtraMetadata(string $path, array $metadata): void
             array_intersect_key($event->getRow(), $columnFilter),
             ['uuid' => $uuid]
         );
+
+        // Unset cache
+        unset($this->records[$path]);
```

Comment:
I think we should probably not throw away the record but update it. Otherwise the next call to `getRecord()` will need to query the database (for the exact same data) and the `RetrieveDbafsMetadataEvent` will be dispatched again.

Something like this (untested):

```suggestion
        // Update cached record
        $this->records[$path]['extra'] = $event->getExtraMetadata();
```

Reply:
Updated :)